src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
Well, yeah, it depends on what you play. If you tap on a some song in the Android app, the entire album will added to tracklist (but only the first is "queued"). If you queue more songs, they will be added individually after the manually queued tracks, but before the automatically added ones. And if you tap on a song again, I think that song will replace the automatically added tracks (or at least move them further down), but the manually queued tracks will play next. I should check this more thoroughly though, maybe I got this a bit wrong...

Replace is called whenever you queue a song manually and I think also if you drag a song up or down in the queue. It might even be used to remove tracks from queue. I'll investigate this tonight.

BTW: my Connect speaker doesn't really support the new queue manipulation features. I can only add one song to queue, which it will play. If I add multiple songs, it will play the last queued next and ignore the rest. The drag handles are not shown in the Android app for this particular device, and I think the remove tick boxes aren't either :cry: